src/features/instance/applications/new/functions/getGithubRepo.ts

```diff
@@ -0,0 +1,119 @@
+function isValidProjectName(name: string) {
+	return /^[a-zA-Z0-9-_]+$/.test(name);
+}
+
+async function getGithubRepo(url: URL) {
+	if (url && url.pathname && url.hostname.includes('github.com')) {
+		const parts = url.pathname.split('/').filter(Boolean);
+
+		try {
+			const response = await fetch(`https://api.github.com/repos/${parts[0]}/${parts[1]}`);
+			const data = await response.json();
+
+			return response.status < 400 ? data.name : null;
+			// eslint-disable-next-line
+		} catch (e) {
+			return null;
+		}
+	}
+}
+
+async function getGithubTags(user: string, repo: string) {
+	const response = await fetch(`https://api.github.com/repos/${user}/${repo}/git/refs/tags`);
+
+	if (response.status < 400) {
+		const tagData = await response.json();
+		return tagData.map((tag: { ref: string }) => tag.ref.split('/').slice(-1)[0]);
+	}
+
+	return [];
+}
+
+async function findNpmPackageName(query: string) {
+	const response = await fetch(`https://registry.npmjs.org/-/v1/search?text=${query}`);
+	const packages = await response.json();
+	const pkg = packages.objects.find((p: { package: { name: string } }) => p.package.name === query);
+
+	return pkg?.package?.name;
+}
+
+async function getNpmDistTags(packageName: string) {
+	// searching for a non-existent package via https://registry.npmjs.org/<packageName> will throw a cors error
+	// so instead, we search for repo using api /search endpoint, compare desired package name
+	// against the returned results array. If one exactly matches, that package exists.
+	// When the package exists, we can then look it up against the registry by its package name (avoiding cors error)
+	// and grab the resulting 'dist-tags' property from the returned payload.
+
+	const packageResponse = await fetch(`https://registry.npmjs.org/${packageName}`);
+	const packageResponseData = await packageResponse.json();
+
+	return packageResponseData['dist-tags'];
+}
+
+interface PackageInput {
+	url: string;
+}
+
+interface PackageMeta {
+	type: 'url' | 'github' | 'npm' | null;
+	user: string | null;
+	repo: string | null;
+	url: string | null;
+	package: string | null;
+	tag: string | null;
+}
+
+function parsePackageType(pkg: PackageInput | null): PackageMeta | null {
+	if (!pkg) {
+		return null;
+	}
+
+	const meta: PackageMeta = {
+		type: null,
+		user: null,
+		repo: null,
+		url: null,
+		package: null,
+		tag: null,
+	};
+
+	if (pkg.url.match('://')) {
```

Comment:
`match()` should only be passed a regexp. You should either use `includes()` or use the regexp `/:\/\//` (which looks a little gross).